SoftTissueSimulation/SoftTissueSimulation.py

```diff
@@ -387,6 +403,21 @@ def stopSimulation(self) -> None:
             browserNode.SetRecordingActive(False)
         self.getParameterNode().Modified()
 
+    def resetSimulation(self) -> None:
+        self.currentStep = 0
+        self.totalSteps = 0
+        #reset mesh
+        points_vtk = numpy_to_vtk(num_array=self.initialgrid, deep=True, array_type=vtk.VTK_FLOAT)
+        vtk_points = vtk.vtkPoints()
+        vtk_points.SetData(points_vtk)
+        self.getParameterNode().modelNode.GetUnstructuredGrid().SetPoints(vtk_points)
```

Comment:
@pieper recommended to not to use this way of updating the mesh as the ownership of the underlying memory by sofa objects and vtk objects can cause this to crash. Instead, you can look at https://github.com/RafaelPalomar/SlicerSOFA/blob/ce4b4131a9ed4a0403641de4cf44efcf455d2c34/MultiMaterialSimulation/MultiMaterialSimulation.py#L289-L298 where I use the approach suggested by @pieper 

As mentioned above, moving as much as possible to the base class will help reusing the code